clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -958,6 +958,9 @@ public void assign(Collection<TopicPartition> partitions) {
             if (timeout < 0)
                 throw new IllegalArgumentException("Timeout must not be negative");
 
+            if (this.subscriptions.hasNoSubscriptionOrUserAssignment())
+                throw new IllegalStateException("Consumer is not subscribed to any topics or assigned any partitions");
```

Comment:
Maybe we should add this exception and the `IllegalArgumentException` above to the javadoc?